public/app/plugins/datasource/loki/querybuilder/parsing.ts

```diff
@@ -432,53 +394,14 @@ function handleQuotes(string: string) {
   return string.replace(/`/g, '');
 }
 
-// Template variables
-// Taken from template_srv, but copied so to not mess with the regex.index which is manipulated in the service
-/*
- * This regex matches 3 types of variable reference with an optional format specifier
- * \$(\w+)                          $var1
- * \[\[([\s\S]+?)(?::(\w+))?\]\]    [[var2]] or [[var2:fmt2]]
- * \${(\w+)(?::(\w+))?}             ${var3} or ${var3:fmt3}
- */
-const variableRegex = /\$(\w+)|\[\[([\s\S]+?)(?::(\w+))?\]\]|\${(\w+)(?:\.([^:^\}]+))?(?::([^\}]+))?}/g;
-
-/**
- * As variables with $ are creating parsing errors, we first replace them with magic string that is parseable and at
- * the same time we can get the variable and it's format back from it.
- * @param expr
- */
-function replaceVariables(expr: string) {
-  return expr.replace(variableRegex, (match, var1, var2, fmt2, var3, fieldPath, fmt3) => {
-    const fmt = fmt2 || fmt3;
-    let variable = var1;
-    let varType = '0';
-
-    if (var2) {
-      variable = var2;
-      varType = '1';
-    }
-
-    if (var3) {
-      variable = var3;
-      varType = '2';
+function getChildWithSelector(node: SyntaxNode, selector: string) {
+  let child: SyntaxNode | null = node;
+  const children = selector.split('.');
+  for (const s of children) {
+    child = child.getChild(s);
+    if (!child) {
+      return null;
     }
-
-    return `__V_${varType}__` + variable + '__V__' + (fmt ? '__F__' + fmt + '__F__' : '');
-  });
-}
-
-const varTypeFunc = [
-  (v: string, f?: string) => `\$${v}`,
-  (v: string, f?: string) => `[[${v}${f ? `:${f}` : ''}]]`,
-  (v: string, f?: string) => `\$\{${v}${f ? `:${f}` : ''}\}`,
-];
-
-/**
- * Get beck the text with variables in their original format.
- * @param expr
- */
-function returnVariables(expr: string) {
-  return expr.replace(/__V_(\d)__(.+)__V__(?:__F__(\w+)__F__)?/g, (match, type, v, f) => {
-    return varTypeFunc[parseInt(type, 10)](v, f);
-  });
+  }
+  return child;
```

Comment:
this is my first time looking at the loki datasource frontend code. is the purpose of `getChildWithSelector` to get the last child in the split selector string?

Reply:
Hmm I guess I should add a comment here. Yeah it's basically just a helper so instead of `node.getChild('foo')?.getChild('bar')?.getChild('baz')` you can write `getChildWithSelector(foo.bar.baz)`. Similar to how some other libraries travers nested jsons for example

Reply:
ah, ok. makes sense. not a big deal, but it might be clearer if were something like `getLastChildWithSelector`